OSS-676: Parallel testing

shestakovg opened the pull request:
### Overview
Adds tests that run about 1k requests to fauna db concurrently.

### Jira ticket
https://faunadb.atlassian.net/browse/OSS-676

Commit:
OSS-676: Added parallel test

parkhomenko commented on FaunaDB.Client.Test/ParallelTest.cs:
```diff
@@ -0,0 +1,112 @@
+        private IList<FaunaClient> GetClientsPool()
+        {
+            List<FaunaClient> clients = new List<FaunaClient>();
+            for (int i =0; i<10; i++)
```
Is there any style checker in this project?
If not, can we add any common style checker to the project?

parkhomenko commented:
Could you add a simple description to the PR, something like I have here:
https://github.com/fauna/faunadb-csharp/pull/140

Commit:
added some whitespaces

Commit:
changed namespace

Commit:
aded extra check for other net platforms

parkhomenko commented on FaunaDB.Client.Test/ParallelTest.cs:
```diff
@@ -0,0 +1,113 @@
+        public async Task SetUpCollectionAsync()
+        {
+            var client = new FaunaClient(secret: faunaSecret, endpoint: faunaEndpoint);
+            Value exist = await client.Query(Exists(Collection(COLLECTION_NAME)));
```
Instead of casting to bool using `(bool)` as you did below, let's do it this way:
```suggestion
            bool exist = (await client.Query(Exists(Collection(COLLECTION_NAME)))).To<bool>().Value;
```

parkhomenko commented on FaunaDB.Client.Test/ParallelTest.cs:
```diff
@@ -0,0 +1,113 @@
+            Value exist = await client.Query(Exists(Collection(COLLECTION_NAME)));
+            
+            //create collection and fill documents
+            if (!(bool)exist)
```
Can we use more explicit names for variables?
`collectionExists` for example sounds better

parkhomenko commented on FaunaDB.Client.Test/ParallelTest.cs:
```diff
@@ -0,0 +1,113 @@
+                    tasks.Add(Task.Run(async () =>await query(clients[random.Next(0, clients.Count-1)])));
+                }
+                Task.WaitAll(tasks.ToArray());
+                Task.Delay(500);
```
Why do we need this delay?

parkhomenko commented on FaunaDB.Client.Test/ParallelTest.cs:
```diff
@@ -0,0 +1,113 @@
+            return sb.ToString();
+        }
+
+        class CollectionDocument
```
Would it be better to call it `SampleDocument for example`?

parkhomenko reviewed (changes requested):
Everything great! I've left a couple of small comments.

Commit:
- review changes
- made separate delegate fro NET Core 2.1. Sometimes getting an errors, that related to HttpConection. It's a core part of framework, so  I can't tuch it

vadimLF reviewed (approved):
LGTM.
